### DISCUSSION TITLE
[Mar 27, 2023 GitHub Outage] `remote: fatal error in commit_refs`

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#15823] [Actions] [answered]
### Story

- I opened an issue about `remote: fatal error in commit_refs` (which I've now reopened as https://github.com/community/community/discussions/51154).

On Mar 27, 2023, GitHub experienced an outage https://www.githubstatus.com/incidents/52z0j6phhnjs.

### Actual Results
```
remote: fatal error in commit_refs
```

### Expected results
A human readable error message explaining what's wrong. It should indicate which category of thing is unhappy
- is there branch protection active? (for this repository, there is no branch protection)
- _I have no idea_

### What happened

GitHub experienced an outage https://www.githubstatus.com/incidents/52z0j6phhnjs

(According to the incident report, it appears to have been resolved in a bit over an hour.)

**Comment by mahdyar** [comment]
Or just like now, they're having issues and I had to check https://www.githubstatus.com/ to see what's wrong!

**Reply by HatemsherifMF** [reply]
i am facing this issue right now also

**Reply by ranjeet94** [reply]
Same here

**Reply by eero-t** [reply]
Cloning and pulling works, pushing fails to `fatal error in commit_refs`

**Reply by FaheemBhatti** [reply]
same here I am also facing this issue

**Reply by FaheemBhatti** [reply]
same here I am also facing this issue

**Reply by JulianJuaner** [reply]
same

**Reply by laurentluc07** [reply]
same for me

**Reply by alsenydiallo** [reply]
Thank god, I thought this was a local issue with my git repo

**Reply by arpecop** [reply]
git tf?

**Reply by peter064226** [reply]
same issue , what's wrong with github ?

**Reply by GeorgiKeranov** [reply]
Same issue, here

**Reply by Illapavan** [reply]
same

**Reply by JulianJuaner** [reply]
the same error makes us meet here lol.

**Reply by anikushalam** [reply]
i am also facing same issue when push code.

**Reply by lerdt** [reply]
Same here! Push/Pull and also Actions refer to the GitHub 500 page.

**Reply by christophechillan** [reply]
same here. I'am facing the same issue

**Reply by shiprab** [reply]
same issue

**Reply by dashy2021** [reply]
same here I am also facing this issue

**Reply by vvelingkar-cci** [reply]
same

**Reply by diallo-bocar** [reply]
same here

**Reply by sauravk** [reply]
same here I am also facing this issue

**Reply by mpp-snehal** [reply]
Same issue.

**Reply by huanhulan** [reply]
same here

**Reply by jeromevvb** [reply]
same here

**Reply by AsmaRasheed99** [reply]
same

**Reply by colin986** [reply]
same

**Reply by zGuilhermeFT** [reply]
same here

**Reply by khushi1351** [reply]
same here

**Reply by inwald-vianai** [reply]
> Cloning and pulling works, pushing fails to `fatal error in commit_refs`

Yes, merging seems ok too

**Reply by dannicada** [reply]
same here

**Reply by antohevinkr** [reply]
Ha here  more people like me

**Reply by MondayCha** [reply]
same here

**Reply by adahadapato** [reply]
having the same issues here, any updates on solutions or is there anything one can do from our end??

**Reply by origami0227** [reply]
same

**Reply by ghost** [reply]
same issue too..i think one just have to wait i guess

**Reply by joaovictorrx** [reply]
same here

**Reply by hemantvraut** [reply]
+1

**Reply by obianujuez** [reply]
Same here. Pushing fails to `fatal error in commit_refs`

**Reply by Sarmed-armis** [reply]
i'm same also 
what happen?

**Reply by anhdungngo1001** [reply]
the same

**Reply by SKGuptaRPI** [reply]
Same here.

**Reply by kirylAlexndr** [reply]
+1

**Reply by sureshmaurya87** [reply]
+1

**Reply by EgorovSV** [reply]
The same

**Reply by UnitedFeodor** [reply]
me too :(

**Reply by Lupescoder** [reply]
the same

**Reply by ctlgavila** [reply]
+1

**Reply by nemanja-dentalmedia** [reply]
Same here :(

**Reply by fbarikzehi** [reply]
same for me

**Reply by symlinktech** [reply]
same here

**Reply by Ednoer** [reply]
same here :(

**Reply by shubhangdixit** [reply]
same here

**Reply by Dedicatus5457** [reply]
+1

**Reply by hanifisenturk** [reply]
Same, I got this error 2 minutes ago. I cannot push anything to the server

**Reply by YuriLukashenko** [reply]
same

**Reply by ManishGotame** [reply]
+1

**Reply by narhd** [reply]
same here

**Reply by pawan-fls** [reply]
same here

**Reply by Sup3r-Us3r** [reply]
Same here

**Reply by sht3756** [reply]
same here

**Reply by VACynthia** [reply]
Oh this is a relief, I thought the problem was on my end.

**Reply by dohy-9443** [reply]
me too!!!

**Reply by esd072181** [reply]
same here

**Reply by kennedybm** [reply]
same, a have a pratical test to deliver today, and I am not able to update the readme

**Reply by YanivWein24** [reply]
same here

**Reply by mbeka02** [reply]
same

**Reply by EduardoBtc** [reply]
same here

**Reply by frappeerpnext** [reply]
I have same problem now

**Reply by terryfri3nds** [reply]
me too

**Reply by hsnbsrn** [reply]
same problem

**Reply by joaovflorisvaldo** [reply]
Same error here, I'm about 30 minutes behind this bug until I found this topic

**Reply by ryank231231** [reply]
same

**Reply by MichaelNotych** [reply]
me too, is there any solution?

**Reply by hidiaa** [reply]
same error here

**Reply by uejosh** [reply]
Same here. I thought it I had messed up something. Relieved that's not the case.

**Reply by joaovflorisvaldo** [reply]
> Eu também, existe alguma solução?
Wait git LOL, https://www.githubstatus.com/ this link is about git status at a glance

**Reply by Nick-Gabe** [reply]
\- OMG I BROKE EVERYTHING
*searches the error*
\- Nah, not my fault

**Reply by PixPunch** [reply]
Some error

**Reply by aelnonym** [reply]
here we go again

**Reply by YanivWein24** [reply]
> here we go again

I added  -f and it worked

**Reply by jsoref** [reply]
https://www.githubstatus.com/incidents/c2jg911dtkjb

**Comment by mesretours** [comment]
Total 8 (delta 7), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
remote: fatal error in commit_refs
 ! [remote rejected] main -> main (failure)
error: failed to push some refs to

**Reply by jolivarjr** [reply]
same here

**Reply by kalpanathanneeru21** [reply]
same here

**Reply by AMKDevTeam** [reply]
same

**Reply by Haseebshah936** [reply]
same

**Reply by PrabhuQA** [reply]
same

**Reply by himaniChaine** [reply]
same

**Reply by MAGGIx1404** [reply]
same

**Reply by cc374-ual** [reply]
same

**Reply by celsopalmeiraneto** [reply]
same

**Reply by ChrisKapteinCM** [reply]
same

**Reply by Marwen94** [reply]
Same

**Reply by RobertBodziony** [reply]
same

**Reply by vimal-360** [reply]
Same here :/

**Reply by Haseebshah936** [reply]
git actions and operations are some issue check this https://www.githubstatus.com/

**Reply by JorgeLAB** [reply]
same

**Reply by vradova** [reply]
Same issue...

**Reply by luciddreams** [reply]
+1

**Reply by Bibhushan336** [reply]
same

**Reply by christfriedbalizou** [reply]
same

**Reply by vimal-360** [reply]
Tried all other solutions like rebase or force push etc.
```
Enumerating objects: 28, done.
Counting objects: 100% (28/28), done.
Delta compression using up to 6 threads
Compressing objects: 100% (20/20), done.
Writing objects: 100% (20/20), 2.16 KiB | 2.16 MiB/s, done.
Total 20 (delta 13), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (13/13), completed with 7 local objects.        
remote: fatal error in commit_refs        
```

**Reply by smiledev1230** [reply]
Same issue but git pull works

**Reply by zGuilhermeFT** [reply]
same here

**Reply by Phuong1310** [reply]
same

**Reply by DatNguyen2106** [reply]
me too

**Reply by wilfredvanessen** [reply]
+1

**Reply by aversan** [reply]
+1

**Reply by mayankmemorang** [reply]
+1

**Reply by mostafamahmoud8** [reply]
same problem here

**Reply by bitnaturo** [reply]
+1

**Reply by MacimeG** [reply]
same here

**Reply by webhive** [reply]
+1

**Reply by FasterThanDisease** [reply]
+ 1

**Reply by logicalperson0** [reply]
Same here

**Reply by ZonicTrout** [reply]
+1
same here

**Reply by amisoni70** [reply]
same here

**Reply by LuisMSAmorim** [reply]
+ 1 same here

**Reply by MuhamedSaudi** [reply]
+1 Same here

**Reply by bayoishola** [reply]
+1

**Reply by zeroitkaran** [reply]
same here

**Reply by tarannumKhan01** [reply]
+1

**Reply by EduardoBtc** [reply]
same here

**Reply by ismailkahraman123** [reply]
+1 :(

**Reply by muhammetalierken** [reply]
+1

**Reply by ITDOGTOR** [reply]
+1

**Reply by nijilrajkr** [reply]
+1

**Reply by AgniVon1** [reply]
+1

**Reply by alan345** [reply]
+1

**Reply by zeeshanibnali** [reply]
++1

**Reply by xupengfe** [reply]
+1

**Reply by RomiImagePlus** [reply]
+1

**Reply by WMadgwickSig** [reply]
+1

**Reply by FernandoAOborges** [reply]
"fatal error in commit_refs", any news?

**Reply by xtiannyeto** [reply]
+1

**Reply by hamidzangiabadi** [reply]
+1

**Reply by hifromkate** [reply]
+1

**Comment by Haseebshah936** [comment]
Github is fixing it github actions and operations are facing some issue you can further check it here https://www.githubstatus.com/. That is the reason why the code is not pushing and showing 

   remote: fatal error in commit_refs

**Reply by yibaben** [reply]
Yeah! I am having the same push issue here

**Reply by Haseebshah936** [reply]
They went down on 12:33 UTC you can check what is your equivalent time zone. If you want to further investigate

**Reply by stivenjc** [reply]
como haci, githud esta caido???

**Comment by yibaben** [comment]
Yeah! I am having the same push issue here

**Reply by Sabahat25** [reply]
me too

**Reply by ReallyBean** [reply]
me too

**Comment by hanstroxler** [comment]
Amen!

**Comment by vidhya211992** [comment]
I am also facing the same issue

**Comment by sandeepdhobi** [comment]
Facing these issue from last 15 minute. :godmode:

**Comment by Yemsquare** [comment]
I'm also experiencing the same issue here, is there any way out

**Comment by alan-agius4** [comment]
Github are having some issues https://www.githubstatus.com/

**Comment by peter064226** [comment]
+1

**Comment by Gaiejj** [comment]
I am also facing the same issue! I am witnessing history 😂

**Comment by SaurabhPatel1997** [comment]
I'm also facing the same issue.

**Comment by Sabahat25** [comment]
me too

**Comment by Gaiejj** [comment]
I am also facing the same issue! I am witnessing history 😂

**Comment by zGuilhermeFT** [comment]
I need commit my code 😭😭😭

**Reply by LoanAlouache** [reply]
You can commit it.... but not push it 😄

**Reply by zGuilhermeFT** [reply]
Yeah, so: I need push my code 😭😭😭

**Reply by sanjayborah15** [reply]
any solution?

**Reply by ner00** [reply]
waiting, that's a solid strategy

**Reply by symlinktech** [reply]
help

**Reply by symlinktech** [reply]
Help

**Reply by amitmall14** [reply]
please help i also facing same problem

**Comment by cb-aditisomani** [comment]
+1

**Comment by Alpeshensar** [comment]
I am also facing the same issue!

**Comment by Alpeshensar** [comment]
I am also facing the same issue!

**Comment by rdpascua** [comment]
facing the same issue

**Comment by NirabhroMakhalLoop** [comment]
same issue

**Comment by ethan-heo** [comment]
How do I know if there is a problem with GitHub?

**Reply by McHardex** [reply]
https://www.githubstatus.com/

**Reply by ethan-heo** [reply]
thank you 👍

**Comment by zaikunzhang** [comment]
I am not sure, but is there any possibility that it is related to https://github.com/orgs/community/discussions/51047#discussioncomment-5437811 ?

**Comment by zaikunzhang** [comment]
I am not sure, but is there any possibility that it is related to https://github.com/orgs/community/discussions/51047#discussioncomment-5437811 ?

**Comment by bigbaldy1128** [comment]
me too

**Comment by bigbaldy1128** [comment]
me too

**Comment by rdpascua** [comment]
<img width="920" alt="image" src="https://user-images.githubusercontent.com/2498246/227945021-b5484171-00d6-48a7-b714-fc7d3bcde9d7.png">

**Comment by SichangHe** [comment]
What the heck. It is like 18 minutes and so many people somehow found this.

**Reply by rikur** [reply]
Probably because nobody can push changes

**Comment by zGuilhermeFT** [comment]
I have the solution...
Up your code via ssh 😂

**Reply by Gurushesh-Metapercept** [reply]
facing same issue

**Reply by Samratmelidous** [reply]
same issue pls help are jaldi thik karo yarr

**Reply by ZonicTrout** [reply]
I may be doing something wrong because I have never used SSH before, but it isn't working for me, and I'm still getting the same error.

**Reply by zGuilhermeFT** [reply]
Now, ssh aren't working to...

**Reply by zGuilhermeFT** [reply]
but as soon as the problem started it was working

**Reply by Gurushesh-Metapercept** [reply]
not working currently

**Comment by manrodrigues** [comment]
Same problem here :(

**Comment by vishalinfowind** [comment]
What's the solution for this? Anybody know?

**Reply by rikur** [reply]
Wait for GitHub to fix it. They finally acknowledged that this is not just "degraded performance" but rather a major outage. See: https://www.githubstatus.com/

**Comment by Sarmed-armis** [comment]
Dears 
i have same issue now!

**Reply by ner00** [reply]
\o/

**Comment by Yegorich555** [comment]
Same issue. It's awful

**Comment by abionics** [comment]
Still same issue
Waiting for fixes in https://www.githubstatus.com/

**Comment by neetaBir** [comment]
I did try a lot of things to solve the issue before I landed on this page ... 😅 
It was not my push issue... but with github itself. 🤣 (15 mins wasted)

**Comment by LuisMSAmorim** [comment]
same issue here

**Reply by RandDGit** [reply]
same here

**Comment by PardoAlejo** [comment]
Same here!!

**Comment by irisdianauy** [comment]
Same here

**Comment by sir-guevara** [comment]
**If you are having this issue** 
`remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
remote: fatal error in commit_refs
! [remote rejected] main -> main (failure)
error: failed to push some refs to`

**Wait for GitHub to fix it. They finally acknowledged that this is not just "degraded performance" but rather a major outage. See**: https://www.githubstatus.com/

**Comment by sir-guevara** [comment]
**If you are having this issue** 
`remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
remote: fatal error in commit_refs
! [remote rejected] main -> main (failure)
error: failed to push some refs to`

**Wait for GitHub to fix it. They finally acknowledged that this is not just "degraded performance" but rather a major outage. See**: https://www.githubstatus.com/

**Comment by irisdianauy** [comment]
Same here

**Comment by Krpiszez** [comment]
finally its nice to know that its not just you :)

**Comment by sureshmaurya87** [comment]
tried lot of things, many stackoverflow answers and finally landed here

**Comment by santoscdavid** [comment]
Same here ; )

**Comment by santoscdavid** [comment]
Same here ; )

**Comment by cjespelet** [comment]
same here!

**Comment by PolisanTheEasyNick** [comment]
same here

**Comment by PolisanTheEasyNick** [comment]
same here

**Comment by rustemib** [comment]
Same here ; )

**Comment by LucasSous** [comment]
háaaaaaaaa

**Comment by AKSHAT2802** [comment]
Same Here
not able to push

**Comment by misha-kolosov-moc** [comment]
same stuff

**Comment by misha-kolosov-moc** [comment]
same stuff

**Comment by z11027747** [comment]
+1

**Comment by Jean1995** [comment]
Help

**Comment by Jean1995** [comment]
Help

**Comment by jsoref** [comment, marked as answer]
Some specific outages by day, if you're seeing this error and the day matches, follow this incident:

Day | GitHub outage incident
-|-
2023-May-9 | https://www.githubstatus.com/incidents/f0mhbz9xn497
2023-Mar-27 | https://www.githubstatus.com/incidents/52z0j6phhnjs

Unfortunately, this discussion was overrun, so I've moved the original discussion to https://github.com/orgs/community/discussions/51154.

**Comment by silverhero13** [comment]
+1

**Comment by lxfu1** [comment]
+1

**Comment by MotooriKashin** [comment]
mark. waiting for solution...

**Comment by o150209001** [comment]
++

**Comment by krisDev000** [comment]
+1

**Comment by George123-ai** [comment]
++++++

**Reply by byju-rosh** [reply]
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

**Reply by George123-ai** [reply]
ahahhahahahahahah , waiting....

**Comment by hubinjie** [comment]
same issue here

**Comment by awaisejaz101** [comment]
waiting for solutions ⌛

**Comment by Ehtesham074** [comment]
I'm waiting for the solution.

**Comment by saravanasai** [comment]
![image](https://user-images.githubusercontent.com/36000435/227947118-3511cfed-449a-4e78-90cc-2d431e45057c.png)



I think the GitHub is down

**Reply by zaxzerv** [reply]
getting the same issue is there any solution here?

**Reply by saravanasai** [reply]
Currently, nope we have to wait.

**Comment by priyamahitha-asl** [comment]
what is the issue? i cannot push the code

**Reply by saravanasai** [reply]
Some issues on GitHub side

**Comment by XindiShang** [comment]
Man! I alomost thought I messed up...

**Reply by navispy** [reply]
have the same issue

**Comment by zaxzerv** [comment]
this is bad right now I can't push

**Comment by chengdev2023** [comment]
same issue. I cannot push

**Comment by ldogimont** [comment]
GitHub incident, then stock market crash and The End.

**Comment by davidemanca-iol** [comment]
+1

**Comment by pgaur3** [comment]
+1

**Comment by nicolas-grzybowski** [comment]
+1

**Comment by SaadEssam** [comment]
same issue

**Comment by CarrKaden** [comment]
Same issue

**Comment by simonllange** [comment]
I have the same issue

**Comment by werlleyg** [comment]
Estou com o mesmo problema.

**Comment by Illapavan** [comment]
same issue from last 30 minutes

**Comment by LoanAlouache** [comment]
Time to take a coffee break guys

**Comment by taleleuma** [comment]
same issue. "remote: fatal error in commit_refs" on multiple repos.  But, the repos those don't have any new commits, no issues. It just happens in case of new commits

**Comment by rafaewaqarqazi** [comment]
Same issue here as well

**Comment by fredwilliamtjr** [comment]
+1

**Comment by zaxzerv** [comment]
wow I'm not alone lol

**Comment by amtray** [comment]
yep same issue here

**Comment by zaxzerv** [comment]
wow I'm not alone lol

**Comment by John-Weak** [comment]
![when-server-down-iceeramen](https://user-images.githubusercontent.com/18065510/227948215-604992ef-99fa-41d2-be66-c10e36aaa37e.gif)

**Comment by Narsinhs** [comment]
I was trying to recreated my token  🗡️

**Comment by GalaxySciTech** [comment]
+1

**Comment by liamhauw** [comment]
same issue

**Comment by mumarejaz** [comment]
damn how long will this last

**Reply by nitin2701** [reply]
same issue

**Comment by abhinavpalakurthy16** [comment]
Im experiencing the same issue, i thought i messed up the code :D

**Comment by rupam-seal** [comment]
! [remote rejected] master -> master (failure)
error: failed to push some refs to

does this error because of github down?

**Reply by fredwilliamtjr** [reply]
Yes

**Reply by rupam-seal** [reply]
do you think the commit that has already been pushed to GitHub but it is not showing on GitHub  will show automatically

**Comment by nangmuathu** [comment]
+1 :( :(

**Comment by juliandierker** [comment]
should work again

**Comment by sabooli** [comment]
Oh, I thought it is My Github that is acting up..

**Comment by fredwilliamtjr** [comment]
Time for coffee, coffee, coffee!

**Comment by radoscis** [comment]
Works already :)

**Comment by ererink** [comment]
it's working now guys

**Comment by shaloo** [comment]
Its back!

**Comment by borkowsk** [comment]
I unexpectedly lost ability to push anything into my repositories!

**Comment by SKGuptaRPI** [comment]
now it's working.

**Reply by sanketjo96** [reply]
yes

**Comment by zaxzerv** [comment]
try this Guys
git pull --rebase origin master
git push -u origin master

**Comment by zeroitkaran** [comment]
![ice-age-manny](https://user-images.githubusercontent.com/88543157/227949281-9a3d3427-b348-471a-8b5f-efe3b5655b9c.gif)

**Comment by SKGuptaRPI** [comment]
Now it's up and working perfectly.

**Comment by chengdev2023** [comment]
working.

**Comment by MichaelNotych** [comment]
working too

**Comment by mariammeqdar** [comment]
Working..

**Comment by yan2441** [comment]
it is working now

**Comment by bishoySouser** [comment]
its working now

**Comment by zeroitkaran** [comment]
its working now 
![Uploading download (1).jpeg…]()

**Comment by rupam-seal** [comment]
working

**Comment by davidemanca-iol** [comment]
Working!

**Comment by amtray** [comment]
working now guys, glad I didn't try anything silly...

**Comment by AntonOshurek** [comment]
working.......niby działa :)

**Comment by sitender-karl** [comment]
Working now

**Comment by joaovictorrx** [comment]
working now! Lets go!

**Comment by akarakus27** [comment]
Problem solved

**Comment by Ehtesham074** [comment]
Hurray!!!  Now, it is working

**Comment by gabilang** [comment]
But I'm getting the same issue now

**Comment by anvinsaji** [comment]
I am getting now

**Reply by danydoerr** [reply]
Same here

**Reply by mikaalanwar** [reply]
Same ...

**Reply by Poweres-K** [reply]
+1

**Reply by levenwindy** [reply]
ERROR： There was an error committing your changes: File could not be edited.

**Reply by wojtekregis** [reply]
Same thing now. According to githubstatus.com, there's an outgage (called _degraded performance_) of GitHub Actions, Operations and Issues. I can't even do a simple push.

**Reply by channyeintun** [reply]
+1

**Reply by pablo-soler** [reply]
+1

**Reply by eidyliveSEO** [reply]
+1

**Reply by sakshisrivastava02** [reply]
Same here

**Reply by pablo-soler** [reply]
Works now

**Reply by acjbizar** [reply]
+1

**Reply by aziz1234** [reply]
+1

**Reply by Sueletting** [reply]
+1

**Reply by VB10** [reply]
+1

**Reply by xlim1997** [reply]
+1

**Reply by kapildave07** [reply]
+1

**Reply by souravpurkait360** [reply]
+1

**Reply by sw-tt-nishilkhatri** [reply]
+1

**Reply by czirjak** [reply]
+1

**Reply by zsofiaelek87** [reply]
+1

**Reply by Marwen94** [reply]
+1

**Reply by nitinjain999** [reply]
remote: fatal error in commit_refs

**Reply by DreasWeiss** [reply]
+1

**Reply by vishnu701** [reply]
+1

**Reply by Nydson** [reply]
**Same error here**
remote: fatal error in commit_refs

**Reply by mayank-sahu-hubilo** [reply]
+1

**Reply by vishnu701** [reply]
it's working now

**Comment by VB10** [comment]
ye, it's working right now.. thank u team

**Comment by alexef** [comment]
I'm also facing this issue on some repos, not all of them.
